thehive/app/org/thp/thehive/services/AlertSrv.scala

```diff
@@ -271,17 +271,26 @@ class AlertSrv @Inject() (
       updatedCase <- mergeInCase(alert, case0)
     } yield updatedCase
 
-  def mergeInCase(alert: Alert with Entity, `case`: Case with Entity)(implicit graph: Graph, authContext: AuthContext): Try[Case with Entity] =
-    for {
-      _ <- caseSrv.addTags(`case`, get(alert).tags.toSeq.map(_.toString).toSet)
-      description = `case`.description + s"\n  \n#### Merged with alert #${alert.sourceRef} ${alert.title}\n\n${alert.description.trim}"
-      c <- caseSrv.get(`case`).update(_.description, description).getOrFail("Case")
-      _ <- importObservables(alert, `case`)
-      _ <- importCustomFields(alert, `case`)
-      _ <- alertCaseSrv.create(AlertCase(), alert, `case`)
-      _ <- markAsRead(alert._id)
-      _ <- auditSrv.alertToCase.merge(alert, c)
-    } yield c
+  def mergeInCase(alert: Alert with Entity, `case`: Case with Entity)(implicit graph: Graph, authContext: AuthContext): Try[Case with Entity] = {
+    auditSrv.mergeAudits {
+      val customFields = get(alert).richCustomFields.toSeq.map(_.toOutput.toJson)
+      val description = `case`.description + s"\n  \n#### Merged with alert #${alert.sourceRef} ${alert.title}\n\n${alert.description.trim}"
+      val tags = get(alert).tags.toSeq.map(_.toString)
+
+      for {
+        _ <- markAsRead(alert._id)
+        _ <- importObservables(alert, `case`)
+        _ <- importCustomFields(alert, `case`)
+        _ <- caseSrv.get(`case`).update(_.description, description).getOrFail("Case")
+        _ <- caseSrv.addTags(`case`, tags.toSet)
+        // No audit for markAsRead and observables
+        // Audits for customFields, description and tags
+        details <- Success(Json.obj("customFields" -> customFields, "description" -> description, "tags" -> tags))
+      } yield details
+    } (details => auditSrv.alertToCase.merge(alert, `case`, Some(details)))
+
+    caseSrv.get(`case`).getOrFail("Case")
```

Comment:
The method `auditSrv.mergeAudits` returns a Try (it could fail), its result must be flatMap with the get case.